internal/jobqueue/job.go

```diff
@@ -120,14 +120,19 @@ func (job *Job) Run() (*common.ComposeResult, error) {
 	for _, t := range job.Targets {
 		switch options := t.Options.(type) {
 		case *target.LocalTargetOptions:
+			filename, _, err := d.FilenameFromType(job.OutputType)
+			if err != nil {
+				r = append(r, err)
+				continue
+			}
 
-			err = runCommand("cp", "-a", "-L", tmpStore+"/refs/"+result.OutputID+"/.", options.Location)
+			f, err := os.Open(tmpStore + "/refs/" + result.OutputID + "/" + filename)
```

Comment:
You mention that this may be slower, would be nice to do a before/after sanity test for how long this takes. My assumption would be that in both cases this would be barely noticeable.

Reply:
I've reworded the commit message and added information that everything is good after this commit. :)